spring-integration-smb/src/test/java/org/springframework/integration/smb/Main.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.smb;
+
+import java.util.Scanner;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.context.support.GenericXmlApplicationContext;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.integration.file.remote.handler.FileTransferringMessageHandler;
+import org.springframework.integration.smb.session.SmbSessionFactory;
+import org.springframework.messaging.support.GenericMessage;
+
+import jcifs.DialectVersion;
+import jcifs.smb.SmbFile;
+
+/**
+ * Starts the Spring Context and will initialize the Spring Integration routes.
+ *
+ * @author Gunnar Hillert
+ * @author Artem Bilan
+ * @author Gregory Bragg
+ *
+ * @since 1.0
+ *
+ */
+public final class Main {
```

Comment:
Why do we need this one?
Why some unit test cannot help?
No one is going to run this class.
Better to have something what is triggered via build tool or on CI pipeline.

Reply:
This Main class is used to verify functionality against a real SMB file share... I tested this against a Dev file server on a Windows VM in AWS, and was able to debug it fully for all functionality that I couldn't debug using JUnit and a Mockito mock session.